src/libs/kata-types/src/config/drop_in.rs

```diff
@@ -213,3 +213,205 @@ mod toml_tree_ops {
+
+    pub fn load(base_cfg_file_path: &Path) -> Result<TomlConfig> {
+        let base_toml_str = fs::read_to_string(&base_cfg_file_path)?;
+        let mut base_config: toml::Value = toml::from_str(&base_toml_str)?;
+        let dropin_dir = get_dropin_dir_path(base_cfg_file_path)?;
+
+        update_from_dropins(&mut base_config, &dropin_dir)?;
+
+        let config: TomlConfig = base_config.try_into()?;
+        Ok(config)
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use std::io::Write;
+
+        fn create_file(path: &Path, contents: &[u8]) -> Result<()> {
+            fs::File::create(path)?.write_all(contents)
+        }
+
+        #[test]
+        fn test_no_dropins_dir() {
+            let tmpdir = tempfile::tempdir().unwrap();
+
+            let runtime_config_file_data = r#"
+                [hypervisor.qemu]
+                path = "/usr/bin/qemu-kvm"
+                default_bridges = 3
+                [runtime]
+                enable_debug = true
+                internetworking_model="tcfilter"
+            "#;
+
+            let mut config_path = PathBuf::from(tmpdir.path());
+            config_path.push("runtime.toml");
+            let result = create_file(&config_path, runtime_config_file_data.as_bytes());
+            assert!(result.is_ok());
+
+            let config = load(&config_path).unwrap();
+            assert_eq!(
+                config.hypervisor["qemu"].path,
+                "/usr/bin/qemu-kvm".to_string()
+            );
+            assert_eq!(config.hypervisor["qemu"].device_info.default_bridges, 3);
+            assert!(config.runtime.debug);
+            assert_eq!(config.runtime.internetworking_model, "tcfilter".to_string());
+        }
+
+        #[test]
+        fn test_no_dropins() {
+            let tmpdir = tempfile::tempdir().unwrap();
+
+            let runtime_config_file_data = r#"
+                [hypervisor.qemu]
+                path = "/usr/bin/qemu-kvm"
+                default_bridges = 3
+                [runtime]
+                enable_debug = true
+                internetworking_model="tcfilter"
+            "#;
+
+            let mut config_path = PathBuf::from(tmpdir.path());
+            config_path.push("runtime.toml");
+            let result = create_file(&config_path, runtime_config_file_data.as_bytes());
+            assert!(result.is_ok());
+
+            let mut dropin_dir = PathBuf::from(tmpdir.path());
+            dropin_dir.push("config.d");
+            let result = fs::create_dir(&dropin_dir);
+            assert!(result.is_ok());
+
+            let config = load(&config_path).unwrap();
+            assert_eq!(
+                config.hypervisor["qemu"].path,
+                "/usr/bin/qemu-kvm".to_string()
+            );
+            assert_eq!(config.hypervisor["qemu"].device_info.default_bridges, 3);
+            assert!(config.runtime.debug);
+            assert_eq!(config.runtime.internetworking_model, "tcfilter".to_string());
```

Comment:
Could use a common helper function for the last piece of the test in both `test_no_dropins_dir` and `test_no_dropins`.